core/src/test/java/kafka/testkit/KafkaClusterTestKit.java

```diff
@@ -150,15 +150,16 @@ public KafkaClusterTestKit build() throws Exception {
                     ThreadUtils.createThreadFactory("KafkaClusterTestKit%d", false));
                 for (ControllerNode node : nodes.controllerNodes().values()) {
                     Map<String, String> props = new HashMap<>(configProps);
-                    props.put(KafkaConfig$.MODULE$.ProcessRolesProp(), "controller");
+                    props.put(KafkaConfig$.MODULE$.ProcessRolesProp(), roles(node.id()));
```

Comment:
Something that was never quite clear to me is why we bother working with the lower-level `ControllerServer` and `BrokerServer` instances in this class. It seems like it would be simpler to use `KafkaRaftServer` which is already doing the dirty work of building `KafkaRaftManager` and the server objects.

Reply:
Yeah, I think that's a good point. I will try to improve this.